pkgs/development/tools/uniffi-bindgen/default.nix

```
@@ -0,0 +1,46 @@
+{ lib
+, stdenv
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, ktlint
+, yapf
+, rubocop
+, rustfmt
+, libiconv
+, makeWrapper
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "uniffi-bindgen";
+  version = "0.17.0";
+
+  src = fetchFromGitHub {
+    owner = "mozilla";
+    repo = "uniffi-rs";
+    rev = "v${version}";
+    hash = "sha256-EGyJrW0U/dnKT7OWgd8LehCyvj6mxud3QWbBVyhoK4Y=";
+  };
+
+  cargoPatches = [ ./cargo-lock.patch ];
```

```suggestion
  cargoLock.lockFile = ./Cargo.lock;
```
Please use the lock file directly.

---

Seems like `cargoLock.lockFile` does not work to generate a lockfile. I changed it to `cargoLock.lockFileContents` and added a `cp` in the `postPatch` as both options do not add a `Cargo.lock`.